### DISCUSSION TITLE
Any way to generate polycrystal(more than 1e4 grains) with low dispersion level of grain size rapidly?

### DISCUSSION THREAD

**Discussion by damulaoshi** [neperfepx/neper#422] [Q&A] [answered]
Hi Romain,
First of all, thanks a lot for your useful software. It really helps me a lot during my research.

The question is, is there any way to generate a polycrystalline model(more than 1e4 grains) with a low dispersion level of grain size rapidly?

I've tried the code by using HPC cluster with 4 nodes, each node has 20 processors, but it still takes more than 9h for generating a single model:
**neper -T -n from_morpho -domain "cube(60,60,60)" -morpho "diameq:lognormal(3.3962,0.005)"  -id 1 -tesrsize 100 -format tesr -statcell vol,diameq,coo  -o test;**

Or even I use Dirac distribution like:
**neper -T -n from_morpho -domain "cube(60,60,60)" -morpho "diameq:dirac(3.3962)"  -id 1 -tesrsize 100 -format tesr -statcell vol,diameq,coo  -o test;**

So, is there any way that I can generate the polycrystalline model (with raster tessellation) with a low dispersion level of grain size more rapidly? Cause I need to generate multiple models, and hoping that there is a way that I can reduce the generating time.

Thanks before!
Best regards,
Sheng

**Comment by rquey** [comment, marked as answer]
Yes, there is.  Generate a centroidal tessellation (and stop iterations early):

```
neper -T -n 10000 -morpho centroidal -morphooptistop "val=1e-2"
neper -V n10000-id1.tess -print img
```
(Runs in less than 1 minute on my laptop).

![image](https://user-images.githubusercontent.com/2659376/160841308-6f608ff2-1fcc-447a-adb0-c353320b5481.png)

Centroidal tessellations are tessellations for which the seeds are brought into coincidence with the cell centers. They have a low dispersion of grain size.

**Reply by damulaoshi** [reply]
Thanks a lot, Romain!
I've tried, and yes, it gives a low dispersion of grain size (coefficient of variation~0.06 for 1.5e4 grains) with less than 3 minutes on my laptop. Solved my problem.
Thanks again!
